packages/flutter_tools/test/general.shard/windows_version_validator_test.dart

```diff
@@ -2,138 +2,39 @@
-System Boot Time:          8/10/2022, 1:03:10 PM
-System Manufacturer:       Google
-System Model:              Google Compute Engine
-System Type:               x64-based PC
-Processor(s):              1 Processor(s) Installed.
-                           [01]: AMD64 Family 23 Model 49 Stepping 0 AuthenticAMD ~2250 Mhz
-BIOS Version:              Google Google, 6/29/2022
-Windows Directory:         C:\\Windows
-System Directory:          C:\\Windows\\system32
-Boot Device:               \\Device\\HarddiskVolume2
-System Locale:             en-us;English (United States)
-Input Locale:              en-us;English (United States)
-Time Zone:                 (UTC-08:00) Pacific Time (US & Canada)
-Total Physical Memory:     32,764 MB
-Available Physical Memory: 17,852 MB
-Virtual Memory: Max Size:  33,788 MB
-Virtual Memory: Available: 18,063 MB
-Virtual Memory: In Use:    15,725 MB
-Page File Location(s):     C:\\pagefile.sys
-Domain:                    ad.corp.google.com
-Logon Server:              \\CBF-DC-8
-Hotfix(s):                 7 Hotfix(s) Installed.
-                           [01]: KB5013624
-                           [02]: KB5003791
-                           [03]: KB5012170
-                           [04]: KB5016616
-                           [05]: KB5014032
-                           [06]: KB5014671
-                           [07]: KB5015895
-Hyper-V Requirements:      A hypervisor has been detected. Features required for Hyper-V will not be displayed.
-''';
+/// A fake [_WindowsUtils] class for a valid version
+class FakeValidOperatingSystemUtils extends Fake
+    implements OperatingSystemUtils {
+  FakeValidOperatingSystemUtils();
 
-const String validWindows11CnStdOut = r'''
-主机名:           XXXXXXXXXXXX
-OS 名称:          Microsoft Windows 11 专业版
-OS 版本:          10.0.22621 暂缺 Build 22621
-OS 制造商:        Microsoft Corporation
-OS 配置:          独立工作站
-OS 构建类型:      Multiprocessor Free
-注册的所有人:     暂缺
-注册的组织:       暂缺
-产品 ID:          XXXXXXXXXXXX
-初始安装日期:     2022/11/9, 13:33:50
-系统启动时间:     2022/11/30, 13:36:47
-系统制造商:       ASUS
-系统型号:         System Product Name
-系统类型:         x64-based PC
-处理器:           安装了 1 个处理器。
-                  [01]: Intel64 Family 6 Model 151 Stepping 2 GenuineIntel ~3600 Mhz
-BIOS 版本:        American Megatrends Inc. 2103, 2022/9/30
-Windows 目录:     C:\WINDOWS
-系统目录:         C:\WINDOWS\system32
-启动设备:         \Device\HarddiskVolume1
-系统区域设置:     zh-cn;中文(中国)
-输入法区域设置:   zh-cn;中文(中国)
-时区:             (UTC+08:00) 北京，重庆，香港特别行政区，乌鲁木齐
-物理内存总量:     65,277 MB
-可用的物理内存:   55,333 MB
-虚拟内存: 最大值: 75,005 MB
-虚拟内存: 可用:   61,781 MB
-虚拟内存: 使用中: 13,224 MB
-页面文件位置:     C:\pagefile.sys
-域:               WORKGROUP
-登录服务器:       \\XXXXXXXXXXXX
-修补程序:         安装了 3 个修补程序。
-                  [01]: KB5020622
-                  [02]: KB5019980
-                  [03]: KB5019304
-Hyper-V 要求:     已检测到虚拟机监控程序。将不显示 Hyper-V 所需的功能。
-''';
+  @override
+  String get name => 'Microsoft Windows [Version 10.0.22621.963]';
```

Comment:
How about making this a constructor parameter? That way, rather than defining different classes, when we want to extend with new test cases, we can just provide a new constructor argument:

```dart
class FakeValidOperatingSystemUtils extends Fake implements OperatingSystemUtils {
  FakeValidOperatingSystemUtils([this.name = 'Microsoft Windows [Version 11.0.22621.963]']);
  
  @override
  final String name;
}
```

Reply:
Great suggestion, I've incorporated this in the next push